projects/cashmere/src/lib/pagination/load-more-pagination.component.spec.ts

```diff
@@ -49,10 +41,9 @@ describe('PaginationComponent', () => {
 
     describe('when _loadNextPage is called on the first page', () => {
         beforeEach(() => {
-            fixture.detectChanges();
-            component.pageNumber = 1;
             component.length = 100;
             component.pageSize = 10;
+            component.pageNumber = 1;
```

Comment:
If we were to add a call to `_sanitizePageNumber()` in the setter for `length`, would this test pass regardless of the order we set the properties?

Reply:
Hmm I suppose it wouldn't pass. I guess I was thinking that once the component is initialized it would be reasonable to require the page number to be set after the other fields. Now that I think about it though, I guess there are cases where the developer doesn't have control over the order in which the inputs are changed.

I just had a thought, what if we were to try to sanitize the page number in `ngOnChanges()`? Could that be a solution? I'm thinking maybe at that point all of the inputs have been changed and we can sanitize confidently.

Reply:
@corykon messed around with this a little more. The ngOnChanges call only seems to happen when the bound values (from the parent component) are changed as opposed to every time the component values are changed. I decided to stick with the original setTimeout approach here, except that instead of reusing the sanitized page number I'm recalculating it after the timeout to account for any other changed inputs. Does this seem good to you?